plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1146,4 +1162,121 @@ protected TestView createSleepingView(Duration minimalSleepDuration)
+        assertQuery(
+                "SELECT regionkey, name FROM " + tableName,
+                "VALUES "
+                        + "(0, 'AFRICA'),"
+                        + "(2, 'ASIA'),"
+                        + "(3, 'EUROPE'),"
+                        + "(4, 'MIDDLE EAST')");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testDeleteWithVarcharEqualityPredicate()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete", "(col varchar(1))", ImmutableList.of("'a'", "'A'", "null"))) {
+            if (!hasBehavior(SUPPORTS_DELETE)) {
+                assertQueryFails("DELETE FROM " + table.getName(), "This connector does not support deletes");
+                return;
+            }
+            if (!hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_EQUALITY)) {
+                assertQueryFails("DELETE FROM " + table.getName() + " WHERE col = 'A'", "Unsupported delete");
+                return;
+            }
+
+            assertUpdate("DELETE FROM " + table.getName() + " WHERE col = 'A'", 1);
+            assertQuery("SELECT * FROM " + table.getName(), "VALUES 'a', null");
+        }
+    }
+
+    @Test
+    public void testDeleteWithVarcharInequalityPredicate()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete", "(col varchar(1))", ImmutableList.of("'a'", "'A'", "null"))) {
+            if (!hasBehavior(SUPPORTS_DELETE)) {
+                assertQueryFails("DELETE FROM " + table.getName(), "This connector does not support deletes");
+                return;
+            }
+            if (!hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_INEQUALITY)) {
+                assertQueryFails("DELETE FROM " + table.getName() + " WHERE col != 'A'", "Unsupported delete");
+                return;
+            }
+
+            assertUpdate("DELETE FROM " + table.getName() + " WHERE col != 'A'", 1);
+            assertQuery("SELECT * FROM " + table.getName(), "VALUES 'A', null");
+        }
+    }
+
+    @Test
+    public void testDeleteWithVarcharGreaterAndLowerPredicate()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete", "(col varchar(1))", ImmutableList.of("'a'", "'A'", "'b'", "null"))) {
+            if (!hasBehavior(SUPPORTS_DELETE)) {
+                assertQueryFails("DELETE FROM " + table.getName(), "This connector does not support deletes");
+                return;
+            }
+            if (!hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_INEQUALITY)) {
+                assertQueryFails("DELETE FROM " + table.getName() + " WHERE col < 'A'", "Unsupported delete");
+                assertQueryFails("DELETE FROM " + table.getName() + " WHERE col > 'A'", "Unsupported delete");
+                return;
+            }
+
+            assertUpdate("DELETE FROM " + table.getName() + " WHERE col < 'A'", 0);
```

Comment:
I know we already discussed it before, but I'm still not convinced to a check where the result is compared with 0.
Up to you.